repos/system_upgrade/common/actors/localrepos/tests/test_localrepos.py

```diff
@@ -0,0 +1,138 @@
+                          ('mirrorlist=file:///some_mirror_list.txt',
+                           'mirrorlist=file:///installroot/some_mirror_list.txt'),
+                          ('gpgkey=file:///etc/pki/some.key',
+                           'gpgkey=file:///etc/pki/some.key'),
+                          ('', ''),
+                          ('[repoid]', '[repoid]')])
+def test_adjust_local_file_url(repo_file_line, expected_adjusted_repo_file_line):
+    adjusted_repo_file_line = localrepos._adjust_local_file_url(repo_file_line)
+    if 'file://' not in repo_file_line:
+        assert adjusted_repo_file_line == repo_file_line
+        return
+    assert adjusted_repo_file_line == expected_adjusted_repo_file_line
+
+
+class MockedFileDescriptor(object):
+
+    def __init__(self, repo_file, expected_new_repo_file):
+        self.repo_file = repo_file
+        self.expected_new_repo_file = expected_new_repo_file
+
+    @staticmethod
+    def _create_repo_string(repo_as_dict):
+        repo = ['[{}]'.format(repo_as_dict['repoid'])]
+        for key in repo_as_dict.keys():
+            if key != 'repoid':
+                repo.append('{}={}'.format(key, repo_as_dict[key]))
+        return '\n'.join(repo)
+
+    def _create_repo_file_string(self, repo_file):
+        repo_strings = []
+        for repo in repo_file:
+            repo_strings.append(self._create_repo_string(repo))
+        return '\n'.join(repo_strings)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        return
+
+    def read(self):
+        return self._create_repo_file_string(self.repo_file)
+
+    def write(self, new_contents):
+        assert self.expected_new_repo_file
+        expected_repo_file_contents = self._create_repo_file_string(self.expected_new_repo_file)
+        assert expected_repo_file_contents == new_contents
+
+
+class MockedContext(object):
+
+    def __init__(self, repo_contents, expected_repo_contents):
+        self.repo_contents = repo_contents
+        self.expected_repo_contents = expected_repo_contents
+
+    def open(self, path, mode):
+        return MockedFileDescriptor(self.repo_contents, self.expected_repo_contents)
+
+
+@pytest.mark.parametrize('repo_file, local_repoids, expected_repo_file',
+                         [(REPO_FILE_1, REPO_FILE_1_LOCAL_REPOIDS, REPO_FILE_1_ADJUSTED),
+                          (REPO_FILE_2, REPO_FILE_2_LOCAL_REPOIDS, REPO_FILE_2_ADJUSTED),
+                          (REPO_FILE_3, REPO_FILE_3_LOCAL_REPOIDS, REPO_FILE_3_ADJUSTED)])
+def test_adjust_local_repos_to_container(repo_file, local_repoids, expected_repo_file):
+    context = MockedContext(repo_file, expected_repo_file)
+    localrepos._adjust_local_repos_to_container(context, '<some_repo_path>', local_repoids)
+
+
+@pytest.mark.parametrize('repo_file', [REPO_FILE_EMPTY, REPO_FILE_1, REPO_FILE_2])
+def test_extract_repos_from_repofile(repo_file):
```

Comment:
I added some test cases for commented out lines, however I'd say that invalid lines shouldn't be present since the files are edited in the target user-space.